routers/web/repo/actions/badge.go

```diff
@@ -19,13 +19,28 @@ import (
 func GetWorkflowBadge(ctx *context.Context) {
 	workflowFile := ctx.PathParam("workflow_name")
 	branch := ctx.Req.URL.Query().Get("branch")
-	if branch == "" {
+	tag := ctx.Req.URL.Query().Get("tag")
+	useLatestTag := ctx.Req.URL.Query().Has("latest_tag")
+	if branch == "" && tag == "" && !useLatestTag {
```

Comment:
This is not right and clear, maybe
```suggestion
  switch {
      case useLatestTag:
      case tag != "":
      case branch != ""
      default:
  }
```